api/data_refinery_api/views.py

```diff
@@ -408,6 +408,12 @@ def get_object(self, pk):
             return Experiment.public_objects.get(pk=pk)
         except Experiment.DoesNotExist:
             raise Http404
+        except Exception:
```

Comment:
Do we actually get here if an accession code is used instead of a ID? It seems like the initial lookup would raise a DoesNotExist which would trigger the previous except block and raise the Http404 before we got here.

Reply:
There are other exceptions than Experiment.DoesNotExist, since we're allowing a more complex input string now.

Reply:
So are you saying an accession code instead of an id will trigger a different exception than DoesNotExist or that you want to be more general about what you catch? I'm thinking what we want to do is remove the except block right above this so this one handles all exceptions, even the DoesNotExist.

Reply:
I changed it from a 404 to a 400 error in the final block, as Bad Request is probably a better explanation for why you'd ever end up there.